oak/server/rust/oak_runtime/src/node/grpc_server.rs

```diff
@@ -0,0 +1,195 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use log::{error, info};
+
+use bytes::Buf;
+use std::{
+    thread,
+    thread::{spawn, JoinHandle},
+};
+use futures_executor::block_on;
+
+use oak_abi::{ChannelReadStatus, OakStatus};
+
+use crate::Handle;
+use crate::{NodeId, RuntimeRef};
+
+trait HttpStatusCode {
+    fn status_code(&self) -> http::StatusCode;
+}
+
+impl HttpStatusCode for ChannelReadStatus {
+    fn status_code(&self) -> http::StatusCode {
+        match self {
+            // TODO: Consider multiple different status codes.
+            ChannelReadStatus::NotReady => http::StatusCode::SERVICE_UNAVAILABLE,
+            ChannelReadStatus::ReadReady => http::StatusCode::OK,
+            ChannelReadStatus::InvalidChannel => http::StatusCode::INTERNAL_SERVER_ERROR,
+            ChannelReadStatus::Orphaned => http::StatusCode::INTERNAL_SERVER_ERROR,
+        }
+    }
+}
+
+impl HttpStatusCode for OakStatus {
+    fn status_code(&self) -> http::StatusCode {
+        match self {
```

Comment:
WDYH about HTTP status codes that we need to return based on the `ChannelReadStatus` and `OakStatus` ?
cc @tiziano88 @daviddrysdale